actionview/lib/action_view/helpers/form_tag_helper.rb

```diff
@@ -442,8 +443,14 @@ def text_area_tag(name, content = nil, options = {})
       #   check_box_tag 'eula', 'accepted', false, disabled: true
       #   # => <input disabled="disabled" id="eula" name="eula" type="checkbox" value="accepted" />
       def check_box_tag(name, value = "1", checked = false, options = {})
+        if checked.is_a?(Hash) && options.empty?
+          options = checked
+          checked = false
+        end
         html_options = { "type" => "checkbox", "name" => name, "id" => sanitize_to_id(name), "value" => value }.update(options.stringify_keys)
-        html_options["checked"] = "checked" if checked
+        if !html_options.key?("checked") && checked
+          html_options["checked"] = "checked"
+        end
```

Comment:
I'm unclear on on the purpose of `!html_options.key?("checked")`.  If `checked` is true, then `html_options["checked"]` should be `"checked"` (or [at least truthy](https://github.com/rails/rails/blob/4491822643f3650f1a7a529d527133bec38b4993/actionview/lib/action_view/helpers/tag_helper.rb#L131-L133)) in any case, no?